translator/internaldata/oc_to_metrics.go

```diff
@@ -250,7 +250,7 @@ func setLabelsMap(ocLabelsKeys []*ocmetrics.LabelKey, ocLabelValues []*ocmetrics
 		lablesCount = len(ocLabelValues)
 	}
 
-	labelsMap.InitEmptyWithCapacity(lablesCount)
+	labelsMap.EnsureCapacity(lablesCount)
```

Comment:
Change in behvaior if `labelsMap` was not empty before. Is this guaranteed? If this is the case I suggest changing setLabelsMap to return  `labelsMap pdata.StringMap` instead of accepting it as a parameter.

Reply:
Returning is not a good pattern for the way we want users to use our pdata. Returning means I need to have a `SetLablesMap` which we don't have because we want users to construct things top-down to avoid extra memory allocation when copying to the parent struct.

In this case, as in all the other cases we start with an empty object during conversion, so there is no problem with having previous values.

Reply:
It is not apparent from setLabelsMap signature that labelsMap is empty. By looking at setLabelsMap code only it is not possible to tell if the code is correct. To know that the code is correct you have to analyse the call site where setLabelsMap.

What do you think about calling Clear() here too (like you want to do in contrib), or declare the requirement in setLabelsMap doccomment that labelsMap must be empty? I think calling Clear() is fine, it should have very little cost.

Reply:
Ok, just to make the reviewer happy :)

Reply:
:-)